halo2-base/src/gates/flex_gate.rs

```diff
@@ -252,7 +330,12 @@ pub trait GateInstructions<F: ScalarField> {
     where
         QA: Into<QuantumCell<F>>;
 
-    /// Returns the inner product of `<a, b>` and the last item of `a` after it is assigned
+    /// Returns the inner product of `<a, b>` and returns a tuple of the last item of `a` after it is assigned and the item to its left `(left_a, last_a)`.
```

Comment:
The tuple returned is (inner product of <a,b>, the last element of `a` now assigned)